packages/chaire-lib-backend/src/services/routing/Routing.ts

```diff
@@ -121,60 +121,77 @@ export class Routing {
         const { isCancelled } = options;
 
         // ** backend
-        const routingResult: TransitOrRouteCalculatorResult[] = [];
+        const routingPromises: Promise<TransitOrRouteCalculatorResult>[] = [];
 
-        for (let i = 0, count = routingAttributes.routingModes.length; i < count; i++) {
+        console.log(`Routing beginning with ${routingAttributes.routingModes.length} modes`);
+        for (const routingMode of routingAttributes.routingModes) {
             if (isCancelled && isCancelled()) {
                 throw 'Cancelled';
             }
+            routingPromises.push(this.calculateRoutingMode(routingMode, routingAttributes));
+        }
 
-            const routingMode = routingAttributes.routingModes[i];
-            try {
-                if (routingMode === 'transit') {
-                    routingResult.push({
-                        routingMode,
-                        result: await calculateTransit(
-                            [routingAttributes.originGeojson, routingAttributes.destinationGeojson],
-                            routingAttributes
-                        )
-                    });
-                } else {
-                    routingResult.push({
-                        routingMode,
-                        result: await calculateRoute(
-                            [routingAttributes.originGeojson, routingAttributes.destinationGeojson],
-                            routingAttributes,
-                            routingMode
-                        )
-                    });
-                }
-            } catch (error) {
-                routingResult.push({
-                    routingMode,
-                    result: !TrError.isTrError(error)
-                        ? new TrError(
-                            `cannot calculate routing for mode ${routingMode}:  ${error}`,
-                            'TRCalculatorError',
-                            { text: 'transit:transitRouting:errors:ErrorForMode', params: { mode: routingMode } }
-                        )
-                        : error
-                });
-            }
+        // Cancel further processing if the request was cancelled
+        if (isCancelled && isCancelled()) {
+            throw 'Cancelled';
```

Comment:
Do we need this both in the loop and after?

Reply:
The one in the loop check cancellation at the beginning, the one outside check after we added all the calls. The idea with cancellation is that you need to check often to interrupt the processing. 
It is really needed, I don't know, we should check in which context it's really used.